akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -574,6 +572,8 @@ private[akka] object Running {
             }
 
             currentState = currentState.applyEvent(setup, event)
+            if (shouldSnapshotAfterPersist == NoSnapshot)
+              shouldSnapshotAfterPersist = setup.shouldSnapshot(currentState.state, event, _currentSequenceNumber)
```

Comment:
it's correct for persist of one event, but not for this persistAll case

Reply:
If that is the case (and I do believe you), are we missing a test case?

Reply:
So, in case of persistAll, we were saying an out-of-date state? 😱

Reply:
I think, only the `shouldSnapshot` predicate did get passed in the old state, before the event was applied.

Reply:
> If that is the case (and I do believe you), are we missing a test case?

Probably not much testing on that detail. I'm afraid I don't have time to add any before my vacation. I'm confident that this is correct and can be merged, but feel free to follow up with more tests.